src/usr/local/www/services_dhcp.php

```diff
@@ -235,6 +241,52 @@ function validate_partial_mac_list($maclist) {
 
 	/* input validation */
 
+	/*
+	 * Check the OMAPI settings
+	 * - Make sure that if the port is defined, that it is valid and isn't in use
+	 * - Make sure the key length is at least 256bit (45 characters)
+	 * - Generate a new key if selected
+	 */
+	if($_POST['omapi_port'] && $_POST['omapi_port'] != "") {
+		// Check the port entry
+		if(in_array($_POST['omapi_port'], range(1, 65535))) {
+			// Get the list of open or listening ports locally
+			exec("/usr/bin/netstat -an4p tcp | awk '{print $4}' | egrep -o '*([0-9]{1,5})$' | sort | uniq", $port_info);
+
+			// Check to see if the port is in the list
+			if(in_array($_POST['omapi_port'], $port_info) && $_POST['omapi_port'] != 7911){
+				$input_errors[] = gettext("Specified port number for OMAPI is in use. Please choose another port or consider using the default.");
+			}
+		} else {
+			$input_errors[] = gettext("The specified OMAPI port number is invalid. Port number must be between 1 and 65635.");
+		}
+
+		// Check the key entry
+		if($_POST['omapi_key'] && $_POST['omapi_key'] != "") {
+			if (strlen($_POST['omapi_key']) < 45) {
+				$input_errors[] = gettext("OMAPI key must be at least 256 bits.");
+			}
+		} elseif($_POST['omapi_gen_key'] == "yes") {
+			//Build the command to generate and capture the key
+			$cmd = "/usr/local/sbin/dnssec-keygen -r /dev/urandom -a HMAC-MD5 -b 512 -n HOST omapi_key; "; //Generate the key
```

Comment:
These should be using an absolute path for the key files. Even if the files are cleaned up after, there is no way to be certain about which directory these are landing in, which could lead to unintended issues.

Reply:
No problem. Added a new commit to address this